docs/docs/migrate-from-v1/README.md

```diff
@@ -0,0 +1,259 @@
+# Migrate from Pipedream v1
+
+The newest version of the Pipedream workflow introduces features like creating code steps in [Python](/code/python), [Bash](/code/bash) and [Go](/code/go).
+
+It also brings the [Node.js](/code/nodes) steps closer to our [Component API](/components/) which powers sources & actions in the platform.
+
+But this also means some of our features have changed. Here's what you need to know.
+
+## New Builder Overview
+
+Fundamentally, the new version of the workflow builder gives you the same abilities to build, test and deploy your workflows. However, there are some differences that change up your normal flow.
+
+### Editing vs Inspecting
+
+When you first open a deployed workflow, you're presented with the **inspector** version of the workflow. In this view you can see logs of past events, and select them to see the results of each step in the workflow.
+
+To edit the workflow, click the **Edit** button in the top right hand corner. This will close the inspector and allow you to edit your workflow without the distraction of logs from the production flow.
+
+### Testing Changes
+
+In the v1 workflow builder to test changes to individual steps required saving then deploying each change to run a test.
+
+We've improved this flow. You can test your changes with a new **Test** button.
+
+In addition to testing single steps, you can now selectively test portions of your workflow in isolation.
+
+<img src="./images/test-workflow-portions.png" alt="Selectively pick testing your workflow above or below the current step is now available.">
+
+Now you can also choose to limit your test runs of your workflow before _or_ after the current step.
+
+### Deploying Changes
+
+After you're happy with your changes, **deploy** them to your production workflow. Just click the **Deploy** button in the top right hand corner of the screen.
+
+After deploying your changes, your workflow is now live.
+
+## Node.js Code Step Changes
+
+There are a few changes to the Node.js code steps that you should know about. Some functions have been renamed for more clarity, and we've aligned the Node.js code steps closer to the [Component API](/docs/components/).
+
+### Code Scaffolding Format
+
+In v1, the Node.js steps would automatically scaffold new Node.js steps in this format:
+
+```javascript
+async (event, steps) {
+  // your code could be entered in here
+}
+```
+
+In v2, the new scaffolding is wrapped with a new `defineComponent` function:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    // your code can be entered here
+  }
+});
+```
+
+1. The `event` from the trigger step is still available, but within `steps.trigger.event` instead.
+2. The `$` variable has been passed into the `run` function where your code is executed.
```

Comment:
Since this is the first time most people will be seeing the `$`, can you link to the component docs on `$` and mention what this means for people? e.g. that you use `$` method to issue HTTP responses, etc.

Reply:
Of course I go over this in my video recording but not in the text